doc/Process_Console.md

```diff
@@ -0,0 +1,595 @@
+ Process Console
+ ===============
+ 
+ `process_console` is a capsule that implements a small shell over the UART that allows
+ a terminal to inspect the kernel and control userspace processes.
+
+  - [Setup](#setup)
+  - [Using Process Console](#using-processconsole)
+  - [Commands](#commands)
+
+ Setup
+ -----
+
+ Here is how to add `process_console` to a board's `main.rs` (the example is taken from the microbit's implementation of the Process console):
+ ```rust
+ static mut PROCESS_PRINTER: Option<&'static kernel::process::ProcessPrinterText> = None;
+ 
+ let process_printer = components::process_printer::ProcessPrinterTextComponent::new()
+        .finalize(components::process_printer_text_component_static!());
+    PROCESS_PRINTER = Some(process_printer);
+
+ let _process_console = components::process_console::ProcessConsoleComponent::new(
+     static_init!(kernel::Kernel, kernel::Kernel::new(&PROCESSES)),
+     components::console::UartMuxComponent::new(
+        &base_peripherals.uarte0,
+        115200,
+        dynamic_deferred_caller,
+    )
+    .finalize(components::uart_mux_component_static!()),
+     components::alarm::AlarmMuxComponent::new(rtc)
+        .finalize(components::alarm_mux_component_static!(nrf52::rtc::Rtc)),
+     process_printer,
+ )
+ .finalize(components::process_console_component_static!(
+        nrf52833::rtc::Rtc
+ ));
+ let _ = _process_console.start();
```

Comment:
Can we just include the simplified version of this? That is, assume we already have `uart_mux` and `alarm_mux` objects?